### DISCUSSION TITLE
Vanilla-extract <> Bun integration?

### DISCUSSION THREAD

**Discussion by dnvt** [vanilla-extract-css/vanilla-extract#1186] [Q&A] [answered]
Hello,
I'm interested in integrating vanilla-extract with Bun. Are there any ongoing discussions or plans regarding this integration/plugin?

Thanks!

**Comment by askoufis** [comment, marked as answer]
While I don't think we've explicitly discussed it, given that bun's plugin API is very similar to esbuild's, I don't think it would be too much effort getting a bun plugin up and running.

**Reply by dragonflyfree** [reply]
Any plans to get this moving? It would be greatly appreciated :)

**Reply by askoufis** [reply]
I briefly tried to port the esbuild plugin to bun, but ran into an issue with one of our dependencies `eval`. It depends on `require-like`, which [uses some node internals](https://github.com/felixge/node-require-like/blob/2206412b5095470de83a2c03c2593ee6315423ab/lib/require-like.js) that bun doesn't support (yet). I think it was `Module._cache` specifically that doesn't work.

I think there was another issue with a difference between bun's plugin API vs esbuild's API, but I don't quite remember.

Happy for someone else to tackle this, not really a high priority for me right now.
